File: GraphleManager/src/main/java/com/graphle/graphlemanager/Connections/RangedNeighborConnections.java
package com.graphle.graphlemanager.Connections;

import com.graphle.graphlemanager.Time.TimeRange;
import org.springframework.lang.NonNull;

import java.util.List;

public record RangedNeighborConnections(@NonNull TimeRange range,
                                        @NonNull List<NeighborConnection> neighborConnections) {
}

File: GraphleManager/src/main/kotlin/com/graphle/graphlemanager/Connections/RangedNeighborConnections.kt
package com.graphle.graphlemanager.Connections

import com.graphle.graphlemanager.Time.TimeRange
import org.springframework.lang.NonNull

@JvmRecord
data class RangedNeighborConnections(
    @field:NonNull @param:NonNull val range: TimeRange,
    @field:NonNull @param:NonNull val neighborConnections: List<NeighborConnection?>
)
